File: tmp-mei/fgen4255027576711118260/base/test/FTestResultKind_0.java
package base.test;
public interface FTestResultKind_0{
FTestResultKind_0 $self = new FTestResultKind_0Impl();
base.test.TestResultKind_0 passed$imm();

base.test.TestResultKind_0 failed$imm(rt.Str details_m$);

base.test.TestResultKind_0 skipped$imm();

base.test.TestResultKind_0 errored$imm(rt.Str details_m$);
static base.test.TestResultKind_0 passed$imm$fun(base.test.FTestResultKind_0 $this) {
  return ((base.test.TestResultKind_0)(switch (1) { default -> {
  System.err.println("Program aborted at:\n"+java.util.Arrays.stream(Thread.currentThread().getStackTrace()).map(StackTraceElement::toString).collect(java.util.stream.Collectors.joining("\n")));
  System.exit(1);
  yield (base.test.TestResultKind_0) null;
}})
);
}

static base.test.TestResultKind_0 skipped$imm$fun(base.test.FTestResultKind_0 $this) {
  return ((base.test.TestResultKind_0)(switch (1) { default -> {
  System.err.println("Program aborted at:\n"+java.util.Arrays.stream(Thread.currentThread().getStackTrace()).map(StackTraceElement::toString).collect(java.util.stream.Collectors.joining("\n")));
  System.exit(1);
  yield (base.test.TestResultKind_0) null;
}})
);
}

static base.test.TestResultKind_0 failed$imm$fun(rt.Str details_m$, base.test.FTestResultKind_0 $this) {
  return ((base.test.TestResultKind_0)(switch (1) { default -> {
  System.err.println("Program aborted at:\n"+java.util.Arrays.stream(Thread.currentThread().getStackTrace()).map(StackTraceElement::toString).collect(java.util.stream.Collectors.joining("\n")));
  System.exit(1);
  yield (base.test.TestResultKind_0) null;
}})
);
}

static base.test.TestResultKind_0 errored$imm$fun(rt.Str details_m$, base.test.FTestResultKind_0 $this) {
  return ((base.test.TestResultKind_0)(switch (1) { default -> {
  System.err.println("Program aborted at:\n"+java.util.Arrays.stream(Thread.currentThread().getStackTrace()).map(StackTraceElement::toString).collect(java.util.stream.Collectors.joining("\n")));
  System.exit(1);
  yield (base.test.TestResultKind_0) null;
}})
);
}
}

File: tmp-mei/fgenjs5030319945734616766/base/test/FTestResultKind_0.js
import * as rt from "../rt/main.js";
export class base$$test$$FTestResultKind_0 {
  static $self = new base$$test$$FTestResultKind_0();

  passed$imm() {
  return base$$test$$FTestResultKind_0.passed$imm$fun(this);
}


failed$imm(details_m$) {
  return base$$test$$FTestResultKind_0.failed$imm$fun(details_m$, this);
}


skipped$imm() {
  return base$$test$$FTestResultKind_0.skipped$imm$fun(this);
}


errored$imm(details_m$) {
  return base$$test$$FTestResultKind_0.errored$imm$fun(details_m$, this);
}

  static passed$imm$fun($this) {
  return (function() {
  console.error("Program aborted at:\n" + new Error().stack);
  if (typeof process !== "undefined") process.exit(1);
  else throw new Error("Program aborted");
})()
;
}

    static skipped$imm$fun($this) {
  return (function() {
  console.error("Program aborted at:\n" + new Error().stack);
  if (typeof process !== "undefined") process.exit(1);
  else throw new Error("Program aborted");
})()
;
}

    static failed$imm$fun(details_m$, $this) {
  return (function() {
  console.error("Program aborted at:\n" + new Error().stack);
  if (typeof process !== "undefined") process.exit(1);
  else throw new Error("Program aborted");
})()
;
}

    static errored$imm$fun(details_m$, $this) {
  return (function() {
  console.error("Program aborted at:\n" + new Error().stack);
  if (typeof process !== "undefined") process.exit(1);
  else throw new Error("Program aborted");
})()
;
}

}
